### PR TITLE
Initialise Puppet module and Ansible role(s) for Springboot apps

### LINKED ISSUE #63: [FEATURE] Create material for deployment tools
* Create Ansible role(s) to deploy  Springboot applications.

* Create Puppet profile(s) to deploy Springboot applications.

### PR DESCRIPTION
**Pull request by hubertqc**
- create the `springboot` profile to manage core structures,
- create the `springboot::app` defined type to configure a single Springboot app.

resolves #63 

Signed-off-by: Hubert Quarantel-Colombani <hubert@quarantel.name>

### CODE REVIEW

File: ansible-roles/ansible-role-springboot_foundation/tasks/lvm.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    vg: "{{ sb_dedicated_vg_opt }}"
+    lv: lv_opt_springboot
+    size: 1g
+    state: present
+  become: "{{ sb_become }}"
+  when: sb_dedicated_lv_opt
+
+- name: "Initialise the XFS filesystem for /opt/springboot"
+  filesystem:
+    fstype: xfs
```
Comment (Laurent-Gaillard) [inline ansible-roles/ansible-role-springboot_foundation/tasks/lvm.yml:14]: fs type and sizes might become parameter in a future enhancement.

File: ansible-roles/ansible-role-springboot_foundation/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+
+- name: "Deploy Springboot foundation"
+  include_tasks: core.yml
+  when:
+    - ansible_os_family == "RedHat"
+    - ansible_distribution_major_version == "7" or ansible_distribution_major_version == "8"  or ansible_distribution_major_version == "9"
```
Comment (Laurent-Gaillard) [inline ansible-roles/ansible-role-springboot_foundation/tasks/main.yml:7]: Why put things for Fedora in galaxy file, ... and not authorize it here?

File: puppet-profiles/springboot/.devcontainer/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# devcontainer
```
Comment (Laurent-Gaillard) [inline puppet-profiles/springboot/.devcontainer/README.md:1]: What is that file for?

File: puppet-profiles/springboot/.devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (Laurent-Gaillard) [inline puppet-profiles/springboot/.devcontainer/devcontainer.json:1]: Same question

**Review by Laurent-Gaillard — DISMISSED** [review]

File: puppet-profiles/springboot/.devcontainer/devcontainer.json (hunk 4 above)
Comment (hubertqc) [reply puppet-profiles/springboot/.devcontainer/devcontainer.json:1]: File created by the PDK convert command, no idea what's used for except making the Puppet module compliant... ;-)

File: puppet-profiles/springboot/.devcontainer/README.md (hunk 3 above)
Comment (hubertqc) [reply puppet-profiles/springboot/.devcontainer/README.md:1]: File created by the PDK convert command, no idea what's used for except making the Puppet module compliant... ;-)

File: ansible-roles/ansible-role-springboot_foundation/tasks/main.yml (hunk 2 above)
Comment (hubertqc) [reply ansible-roles/ansible-role-springboot_foundation/tasks/main.yml:7]: Was a bit lazy to implement the Fedora tests, but will...

File: ansible-roles/ansible-role-springboot_foundation/tasks/lvm.yml (hunk 1 above)
Comment (hubertqc) [reply ansible-roles/ansible-role-springboot_foundation/tasks/lvm.yml:14]: Surely will, later ;-)

File: ansible-roles/ansible-role-springboot_foundation/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+> **Disclaimer:**
+>
+> - The stable versions of the role are tagged. You can find all the available tags with their release notes under Repository/Tags. When you import the role in the requirements.yml file we recommend to use a tagged version of the role.
+> - You may find the release notes & the a migration guide for the latest versions in the [release-notes.md](release-notes.md) file.
+
+Requirements
+------------
+
+- The Springboot application services will being handled as instances of a SystemD template service (not if `sb_become` is set on no/false).
```
[the remark this answers is not in the document]
Comment (hubertqc) [reply ansible-roles/ansible-role-springboot_foundation/README.md:14]: Good point

File: ansible-roles/ansible-role-springboot_foundation/tasks/directories.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    owner: root
+    group: springboot
+    seuser: system_u
+    serole: object_r
+    setype: bin_t
+  become: "{{ sb_become }}"
+
+- name : "Restore SELinux fcontexts for Springboot roots (/opt/springboot /var/log/springboot)"
+  shell: >
+        /bin/bash -c "/bin/chcon -R -u system_u  /opt/springboot /var/log/springboot "/srv/springboot" ; /sbin/restorecon -Rvi /opt/springboot /var/log/springboot "/srv/springboot""
```
[the remark this answers is not in the document]
Comment (hubertqc) [reply ansible-roles/ansible-role-springboot_foundation/tasks/directories.yml:53]: Will fix this

File: ansible-roles/ansible-role-springboot_foundation/tasks/directories.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+- name: Create Springboot APPs common bin directory
+  file:
+    path: "/opt/springboot/bin"
+    state: directory
+    mode: 0555
+    owner: root
+    group: springboot
+    seuser: system_u
+    serole: object_r
+    setype: bin_t
```
[the remark this answers is not in the document]
Comment (hubertqc) [reply ansible-roles/ansible-role-springboot_foundation/tasks/directories.yml:48]: Good point

**Pull request merged**